app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -1006,7 +1006,7 @@ abstract class BaseBrowserFragment :
                 .collect {
                     if (!onboarding.userHasBeenOnboarded() &&
                         it.content.loadRequest?.triggeredByRedirect != true &&
-                        it.source !in intentSourcesList &&
+                        it.source is SessionState.Source.External &&
```

Comment:
This should be `!is`, i.e., not from an intent.

Reply:
To keep the same behaviour (which we should likely refactor another time): Don't finish onboarding the tab was opened from an external source / intent.

Reply:
One change here would be that after restoring an external tab we would still not finish onboarding. Before this change it would be `Source.Restored` whereas now we restore the external source. Not sure if worth it but we could make it possible to check for both `External` and `Restored`?